### DISCUSSION TITLE
connect lighthouses to each other

### DISCUSSION THREAD

**Discussion by hmoffatt** [slackhq/nebula#1079] [Q&A] [answered]
I'd like to run two lighthouses for redundancy, and the two nodes need to be able to communicate with each other as well.

The sample configuration says that the `hosts` list should be empty on lighthouse nodes. How do I connect my two lighthouses together?

**Comment by johnmaguire** [comment]
@hmoffatt Lighthouses do not connect to each other. They operate independently. For two nodes to reach each other, they need at least one Lighthouse in common.

**Reply by hmoffatt** [reply]
Since I want to have two or more lighthouses, but also have those nodes use the overlay network like any other (ie they will not be dedicated to running the lighthouse), would it be useful to run two instances of nebula on them, one as lighthouse with `tun.disabled: true`, and one as a non-lighthouse?

**Reply by johnmaguire** [reply]
@hmoffatt you can do that if you like. You can also run them with tun enabled if you prefer - as long as they appear in each other's static_host_map they should still be able to form a connection like any other Nebula node. They just won't reach out for Lighthouse results.

**Reply by hmoffatt** [reply]
Ah so the nodes will be able to communicate if they are both in each other's static_host_map, even though they don't share a lighthouse?

**Reply by johnmaguire** [reply, marked as answer]
Yep!

**Reply by hmoffatt** [reply]
Great, thanks for your assistance!

**Reply by zoechi** [reply]
> @hmoffatt you can do that if you like. You can also run them with tun enabled if you prefer - as long as they appear in each other's static_host_map they should still be able to form a connection like any other Nebula node. They just won't reach out for Lighthouse results.

When the two lighthouses are on each dide of NAT, I can't add the inside one to static_host_map of the outside one because there is no routeable IP.
How could this be made to eork?

Now I can reach each inside host from outside, except the one that is also a lighthouse.

**Reply by johnmaguire** [reply]
@zoechi Lighthouses should never be placed behind a NAT. They should be Internet routable. You must setup port forwards.

**Reply by zoechi** [reply]
The idea was to have one lighthouse inside my private network and one exposted to the internet.
The former should keep the Nebula network working when the Internet is down and the later to allow access from outside the private network. 
I think this is a reasonable configuration.

The problem seems to be that the internal lighthouse is also a network node. Somehow this configuration seems not to be supported or I'm just missing some understanding how to set it up properly.
A workaround I can think of is to have two Nebula instances on the same machine, where one acts only as lighthouse and the other instance only acts as node.

**Reply by johnmaguire** [reply]
@zoechi If I'm following, your desire is to use the host as a Lighthouse for only a subset of the network, while continuing to use it as a regular node with the entire network. This is not a scenario that was in mind during Nebula development. The issue here is that Lighthouses do not report their addresses / check in with other Lighthouse nodes.  I agree that your best bet in this situation is to run two Nebula instances with separate Nebula IPs. You can run the Lighthouse with tun disabled.

**Reply by zoechi** [reply]
Thanks for confirming. Now I feel more confident about my understanding about how Nebula works.

**Comment by migs35323** [comment]
heh, i am gonna steal this thread instead of creating a basically the same one!!

what are the issues created when i have 2 lighthouses but they are the host at each other?

example:
lighthouse1:
```
static_host_map:
  192.168.100.101: [ 1.1.1.1 ] # lighthouse 1
  192.168.100.102: [ 1.1.1.2 ] # lighthouse 2
lighthouse:
  am_lighthouse: true
  hosts: [ "192.168.100.102" ]

``` 
lighthouse2:
```
static_host_map:
  192.168.100.101: [ 1.1.1.1 ] # lighthouse 1
  192.168.100.102: [ 1.1.1.2 ] # lighthouse 2
lighthouse:
  am_lighthouse: true
  hosts: [ "192.168.100.101" ]

``` 


i am trying to make a HA network, where i can safetly put one of the lighthouse down or spawn with other addresses and nothing gets lost,
and both of the lighthouses need to be able to see each other
sadly i don't think there is a way for one lighhouse share the list with each other right? (so one host is connected to 1 lighthouse but can still talk with another host only connected to the other)

**Reply by hmoffatt** [reply]
On the lighthouse nodes, I have `hosts:` empty (the sample config explicitly says that the hosts list should be empty on lighthouse nodes). On the other nodes I list all the lighthouses.
